### PR TITLE
Add new commands

### PR DESCRIPTION
**Pull request by lukeeey**
# Moved to #333

---

This pull request aims to add many commands from vanilla Minecraft and have the syntax and messages as close as possible.

### Changes
* Added `EntityLiving.hasEffects()` to return whether or not an entity has active effects
* Added `EntityPlayer.setOp( boolean value )`
* Added `EntityPlayer.sendTitleAnimationTimes( float fadeIn, float duration, float fadeOut )`
* Renamed a few constants in the `PotionEffect` enum to make it easier for the /effect command
* Added `PotionEffect.getEffectName()` to return the human readable name from the constant
* Added `Gamemode.getGamemodeName()` to return the human readable name
* Added `World.getDifficulty()`
* Added `World.setDifficulty( Difficulty difficulty )`
* Changed `EntityPlayer.set/getLevel()` to `set/getXPLevel()`

### Commands
- [x] `/op <player>`
- [x] `/deop <player>`
- [x] `/title`
  - [x] `/title <player> clear`
  - [x] `/title <player> reset`
  - [x] `/title <player> <title|subtitle|actionbar> <text>`
  - [x] `/title <player> times <fadeIn> <duration> <fadeOut>`
- [x] `/effect`
  - [x] `/effect <player> clear`
  - [x] `/effect <player> <effect> [seconds] [amplifier] [hideParticles]`
- [x] `/gamemode <mode> [player]` 
- [x] `/difficulty <difficulty>` 
- [x] `/list`   (changed to be exactly like vanilla)
- [ ] `/give <player> <item> [amount] [data] [components]` (doesnt seem to be an api to do it yet)  
- [x] `/tell <target> <message>`
- [ ] `/xp <amount> [player]` (basic implementation but need postfixes)
- [ ] `/stopsound [player] [sound]` (seems to be missing a parameter? needs looking in to)
- [ ] `/playsound [sound] [player] [pitch] [volume] [minimumVolume]` (overloads seem to be in the wrong order)

### Follow up
* Remove gamemode name and potion effect name from the API?

### CODE REVIEW

**Comment by Kaooot** [comment]
Please add /msg <player> <message> and /give <player> <itemStack> <amount> too. :)

File: s [hunk 1]
```diff
@@ -0,0 +1,125 @@
+warning: LF will be replaced by CRLF in gomint-api/src/main/java/io/gomint/world/Gamemode.java.
```
Comment (lukeeey) [inline s:1]: Lmao wtf happened here

File: gomint-server/src/main/java/io/gomint/server/GoMintServer.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 package io.gomint.server;
 
+import com.google.common.util.concurrent.Futures;
```
Comment (lukeeey) [inline gomint-server/src/main/java/io/gomint/server/GoMintServer.java]: Wtf intellij

File: gomint-server/src/main/java/io/gomint/server/command/internal/TitleCommand.java [hunk 3]
```diff
@@ -64,17 +65,16 @@ public CommandOutput execute( CommandSender player, String alias, Map<String, Ob
         if( arguments.containsKey( "type" ) ) {
             switch( (String) arguments.get( "type" ) ) {
                 case "title":
-                    target.sendTitle( (String) arguments.get( "titleText" ) );
+                    target.sendTitle( (String) arguments.get( "text" ) );
                     break;
                 case "subtitle":
-                    target.sendTitle( null, (String) arguments.get( "text" ) );
+                    target.sendTitle( "", (String) arguments.get( "text" ) );
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/command/internal/TitleCommand.java]: don't you overwrite the title now?

Comment (lukeeey) [reply gomint-server/src/main/java/io/gomint/server/command/internal/TitleCommand.java]: @Janmm14 https://github.com/GoMint/GoMint/pull/301/files#diff-930bb11bbb10af48ae040e72a977a72eR1506

Comment (lukeeey) [reply gomint-server/src/main/java/io/gomint/server/command/internal/TitleCommand.java]: @Janmm14 https://github.com/lukeeey/GoMint/blob/424fd579d31fb8f435122e1d614c71f427b42f92/gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java#L1506

**Comment by lukeeey** [comment]
@geNAZt

**Comment by lukeeey** [comment]
I dont get why it failed

**Comment by lukeeey** [comment]
im gonna remake this PR because i messed it up lol

**Pull request closed without merging**